javascript/javascript1/week4/index.js

```diff
@@ -0,0 +1,131 @@
+"use strict"
+let personName;//declared here for global scope.
+const todoArray = [];
+
+function getReply(command){
+    if(typeof(command)!= "string"){
+     return 'please enter a valid command'       
+    }
+const arrayOfCommand = command.split(' '); 
+const todo = command.substring(4,command.length-11); // extracting the characters for pushing todoactivity 
+const firstNumber = Number(arrayOfCommand[2]); //converting into numbers for mathematical operation
+const secondNumber = Number(arrayOfCommand[4]);
+
+
+     switch (command){
+         case 'Hello my name is Rajani Busani':
+             personName = arrayOfCommand[4];
+               return `Nice to meet you ${personName}`; 
+             break;           
+         case 'What is my name?':
+             if(personName){
+               return 'your name is ' + personName+'.'; 
+             }else{
+               return'please enter your name';
+             }
+             break;
+         case 'Add fishing to my todo':                               
+             todoArray.push(todo);
+               return `${todo} added to your todo.`;
+             break;
+         case 'Add singing in the shower to my todo':
+             todoArray.push(todo);             
+               return todo+' added to your todo.';
+             break;
+         case 'Add buying the flowers to my todo':                               
+             todoArray.push(todo);
+               return `${todo} added to your todo.`;
+             break;
+         case 'Remove fishing from my todo':        
+              return removeTodoActivity(command)
+             break;
+         case 'What is on my todo?':
+              const todoString  = todoArray.join(' and ');              
+               return `You have ${todoArray.length} todos - ${todoString}.`
+             break;
+           case 'What day is it today?':
+               const today = new Date();
+               const date = today.getDate();
+               const month = today.getMonth();
+               const months = ["January","February","March","April","May","June","July",
+               "August","September","October","November","December"];
+               const year = today.getFullYear();
+                 return `${date}th of ${months[month]}, ${year}`; 
+               break;
+           case 'what is 3 + 3':
+               if (arrayOfCommand[3]=== '+'){
+                   const sum = firstNumber+secondNumber; 
+                   return sum;                  
+               }
+               break;
+            case 'what is 4 * 12':
```

Comment:
it can be made more dynamic like below  for example by converting to array of words

if (command.startsWith("what is")) {
    const commandWords = command.split(" ");
    const number1 = parseInt(commandWords[2]);
    const number2 = parseInt(commandWords[4]);
    const operation = commandWords[3];
    if (number1 && number2) {
      switch (operation) {
        case "+":
          return number1 + number2;
        case "-":
          return number1 - number2;
        case "*":
          return number1 * number2;
        case "/":
          return number1 / number2;
      }